js/demo-app/src/index.js

```diff
@@ -41,17 +41,37 @@ registerElements();
 
 // ----- Client/Nav Setup
 
-let router = document.getElementById("router");
-router.registerClients(
-  NAV_CONFIGS.reduce((clientMap, {id, url, assignedRoute}) => {
+// Build client list from nav entries and background tasks
+let allClients = {
+  ...NAV_CONFIGS.reduce((clientMap, {id, url, assignedRoute}) => {
     clientMap[id] = {
       url,
       assignedRoute
     }
 
     return clientMap;
-  }, {})
-);
+  }, {}),
+  ...{
+    // Standard worker
+    'backgroundWorker1': {
```

Comment:
This is the first shot at the API for registering a background worker.  Just a simple boolean and a url is all that's necessary.  These types are tracked in typescript as a union type, so I didn't add a type property at this point, however, I could easily switch to that.  Thoughts?